ruma-api-macros/src/util.rs

```diff
@@ -65,23 +65,19 @@ pub(crate) fn request_path_string_and_parse(
                 |(i, segment)| {
                     let path_var = &segment[1..];
                     let path_var_ident = Ident::new(path_var, Span::call_site());
-
-                    let decoded = request::try_deserialize(quote! {
-                        ruma_api::exports::percent_encoding::percent_decode(segment)
-                            .decode_utf8()
-                    });
-                    let path_var = request::try_deserialize(
-                        quote! { std::convert::TryFrom::try_from(decoded.deref()) },
-                    );
                     quote! {
                         #path_var_ident: {
                             use std::ops::Deref as _;
                             use ruma_api::error::RequestDeserializationError;
 
                             let segment = path_segments.get(#i).unwrap().as_bytes();
-                            let decoded = #decoded;
+                            let decoded = ::ruma_api::try_deserialize!(
+                                request, 
+                                ruma_api::exports::percent_encoding::percent_decode(segment)
```

Comment:
If you want to write `::somecrate::` instead of `somecrate::` in new code, please do that consistently.

Reply:
Would it make sense to change everything to `::crate::` so it's consistent across the ruma workspace? That would be it's own PR.

Reply:
Yeah, that would be nice.